src/main/java/net/minecraftforge/registries/RegistryManager.java

```diff
@@ -163,6 +165,15 @@ static <V extends IForgeRegistryEntry<V>> void registerToRootRegistry(ForgeRegis
             registry.register(forgeReg.getRegistryKey(), wrapper, Lifecycle.experimental());
     }
 
+    @SuppressWarnings("unchecked")
+    static <V extends IForgeRegistryEntry<V>> void registerToBuiltinRegistry(ForgeRegistry<V> forgeReg)
+    {
+        WritableRegistry<Registry<V>> registry = (WritableRegistry<Registry<V>>) BuiltinRegistries.REGISTRY;
+        Registry<V> wrapper = forgeReg.getWrapper();
+        if (wrapper != null)
+            registry.register(forgeReg.getRegistryKey(), wrapper, Lifecycle.experimental());
+    }
```

Comment:
Instead of code duplication, probably better to extract the last 3 lines of both functions, and possibly the unchecked cast too, to a private static helper function.